src/command/list/human.rs

```diff
@@ -41,14 +43,68 @@ pub(super) fn format(toolchain: &Toolchain) -> Option<String> {
 /// Accepts the components *from* the toolchain rather than the item itself so
 /// that
 fn display_active(
-    node: &Option<Node>,
-    package_manager: &Option<PackageManager>,
+    runtime: &Option<Box<Node>>,
+    package_manager: &Option<Box<PackageManager>>,
     packages: &[Package],
 ) -> String {
-    match (node, package_manager, packages) {
+    match (runtime, package_manager, packages) {
         (None, _, _) => NO_RUNTIME.to_string(),
-        (Some(node), Some(package_manager), packages) => unimplemented!(),
-        _ => unimplemented!(),
+        (Some(runtime), Some(package_manager), packages) => {
+            let width = text_width().unwrap_or(0);
+            let runtime_version: String = Wrapper::new(width)
+                .initial_indent(INDENTATION)
+                .fill(&format!("Node: {}", list_runtime(runtime)));
+            let package_manager_version: String = Wrapper::new(width)
+                .initial_indent(INDENTATION)
+                .fill(&format!("Yarn: {}", list_package_manager(package_manager)));
+            let package_versions = if packages.is_empty() {
+                Wrapper::new(width)
+                    .initial_indent(INDENTATION)
+                    .fill(&format!("Tool binaries available: NONE"))
+            } else {
+                Wrapper::new(width)
+                    .initial_indent(INDENTATION)
+                    .subsequent_indent(INDENTATION)
+                    .fill(&format!(
+                        "Tool binaries available:\n{}",
+                        display_tool_sets(packages)
+                    ))
+            };
+
+            format!(
+                "⚡️ Currently active tools:\n\n{}\n{}\n{}\n\n{}",
+                runtime_version,
+                package_manager_version,
+                package_versions,
+                "See options for more detailed reports by running `volta list --help`."
+            )
+        }
+        (Some(runtime), None, packages) => {
+            let width = text_width().unwrap_or(0);
+            let runtime_version: String = Wrapper::new(width)
+                .initial_indent(INDENTATION)
+                .fill(&format!("Node: {}", list_runtime(runtime)));
+            let package_versions = if packages.is_empty() {
+                Wrapper::new(width)
+                    .initial_indent(INDENTATION)
+                    .fill(&format!("Tool binaries available: NONE"))
+            } else {
+                Wrapper::new(width)
+                    .initial_indent(INDENTATION)
+                    .subsequent_indent(INDENTATION)
+                    .fill(&format!(
+                        "Tool binaries available:\n{}",
+                        display_packages(packages)
+                    ))
+            };
```

Comment:
`display_packages` looks like it already has internal logic for detecting an empty package list, is it possible to leverage that instead of duplicating this block?

Reply:
Actually this one seems quite tricky to me 😄 So use `display_packages` as an example:

- `display_packages` will be called to display both `Toolchain::Packages` and `Toolchain::Active`.

- For `Toolchain::Packages`, if it's empty, then it should show:

```
⚡️ No tools or packages installed.

You can safely install packages by running `volta install <package name>`.
See `volta help install` for details and more options.
```

- For `Toolchain::Active`, if it's empty, then it should show:

```
Tool binaries available: NONE
```

That's the main reason why I have the double empty check, to handle these two different logics, but I totally agree that we should make the code more clear. I will take another look to see how I could change it.